### PR TITLE
Add per-medium breakouts to dashboard statistics inventories. (PP-728)

### PR DESCRIPTION
**Pull request by tdilauro**
## Description

- Add per-medium breakouts to dashboard statistics inventories.
- Add supporting tests and clean up some tests.
- Resolve some PEP-8 and mypy issues with SQLAlchemy filters.

This work requires changes to the circulation admin UI. The following work is required before this PR can be moved out of `draft` status and merged:
- [X] Change interface in `circulation-admin` and make the changes required for PP-728. 
  These changes are introduced in [circulation-admin PR 98](https://github.com/ThePalaceProject/circulation-admin/pull/98).
- [x] Release a [new version (v1.12.0)](https://github.com/ThePalaceProject/circulation-admin/releases/tag/v1.12.0) of `circulation-admin`.
- [X] Update this PR with new version of `circulation-admin`.

Note:
- Based on testing, the changes in this PR are backward compatible with recent versions of the circulation admin UI.
- The API has changed, so I am labeling as _incompatible changes_.
## Motivation and Context

Need to support additional detail by type of medium (book, audiobook, etc.).

[Jira [PP-728](https://ebce-lyrasis.atlassian.net/browse/PP-728)]

## How Has This Been Tested?

- Additional tests.
- Manual testing in local development environment.
- [CI tests for associated branch](https://github.com/ThePalaceProject/circulation/actions/runs/7081348146) pass.

## Checklist

- [X] I have updated the documentation accordingly.
- [X] All new and existing tests passed.

[PP-728]: https://ebce-lyrasis.atlassian.net/browse/PP-728?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jonathangreen — APPROVED** [review]
Might be a couple lines of commented out testing code that should get cleaned up before merging, but otherwise looks good to me! 🎸 

I also left a couple minor comments, but as always feel free to make those changes or just merge as is.

File: api/admin/dashboard_stats.py [hunk 1]
```diff
@@ -16,25 +15,37 @@
...
+
+if TYPE_CHECKING:
+    from sqlalchemy.sql.expression import ColumnElement
 
 
 def generate_statistics(admin: Admin, db: Session) -> StatisticsResponse:
     return Statistics(db).stats(admin)
 
 
 class Statistics:
```
[the remark this answers is not in the document]
Comment (tdilauro) [reply api/admin/dashboard_stats.py:44]: Good idea! I used it as an opportunity to group some functionality and hopefully make the mainline code a little more understandable.

File: api/admin/dashboard_stats.py [hunk 2]
```diff
@@ -51,55 +62,118 @@ def _libraries_for_admin(self, admin: Admin) -> list[Library]:
...
-                func.sum(LicensePool.licenses_owned),
-                func.sum(LicensePool.licenses_available),
-            )
-            .filter(collection_filter)
-            .filter(self.METERED_LICENSE_FILTER)
-            .all()[0],
+        _query_stats_group: Callable = partial(
+            self._collection_statistics_by_medium_query, collection_filter
+        )
+        return dict(
```
[the remark this answers is not in the document]
Comment (tdilauro) [reply api/admin/dashboard_stats.py]: Good idea. I used it as an opportunity to group some functionality and hopefully make the mainline code a little more understandable.

File: tests/api/admin/test_dashboard_stats.py [hunk 3]
```diff
@@ -325,23 +320,36 @@ def test_stats_collections(admin_statistics_session: AdminStatisticsSessionFixtu
...
     response = session.get_statistics()
-    assert (
-        expected_library_inventory
-        == response.libraries_by_key.get(default_library.short_name).inventory_summary
-    )
+    library_stats_data = response.libraries_by_key.get(default_library.short_name)
+    # all_collections_by_id = {c.id: c for c in response.collections}
+    # library_collections_by_id = {
+    #     id_: all_collections_by_id[id_] for id_ in library_stats_data.collection_ids
+    # }
```
[the remark this answers is not in the document]
Comment (tdilauro) [reply tests/api/admin/test_dashboard_stats.py]: From some refactoring. Cleaned up.

File: api/admin/dashboard_stats.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tdilauro) [reply api/admin/dashboard_stats.py:44]: I went ahead and did this, which led to needing some more hinting. But nice to have that all in order.

**Comment by tdilauro** [comment]
@jonathangreen I made some substantial changes while following your suggestion to use a `dataclass`, so you might want to take another look. No worried if you don't and no rush, as I won't be merging until the admin UI is updated.

Also, codecov seems to have lost its mind. I just manually verified that the tests were hitting lines that it claimed they were missing. 🤷🏽‍♂️

**Review by jonathangreen — APPROVED** [review]
Changes look great!

**Pull request merged**